subsys/bluetooth/shell/flash.c

```diff
@@ -192,31 +198,49 @@ static int cmd_test(int argc, char *argv[])
 	while (repeat--) {
 		result = flash_erase(flash_dev, addr, size);
 		if (result) {
-			printk("Erase Failed, code %d.\n", result);
+			error(shell, "Erase Failed, code %d.", result);
 			return -EIO;
 		}
 
-		printk("Erase OK.\n");
+		print(shell, "Erase OK.");
 
 		if (flash_write(flash_dev, addr, test_arr, size) != 0) {
-			printk("Write internal ERROR!\n");
+			error(shell, "Write internal ERROR!");
 			return -EIO;
 		}
 
-		printk("Write OK.\n");
+		print(shell, "Write OK.");
 	}
 
-	printk("Erase-Write test done.\n");
+	print(shell, "Erase-Write test done.");
 
 	return 0;
 }
 
-static const struct shell_cmd flash_commands[] = {
-	{ "flash-write", cmd_flash, "<address> <Dword> <Dword>..."},
-	{ "flash-erase", cmd_erase, "<page address> <size>"},
-	{ "flash-read", cmd_read, "<address> <Dword count>"},
-	{ "flash-test", cmd_test, "<address> <size> <repeat count>"},
-	{ NULL, NULL, NULL}
+#define HELP_NONE "[none]"
+
+SHELL_CREATE_STATIC_SUBCMD_SET(flash_cmds) {
+	SHELL_CMD(write, NULL, "<address> <dword> <dword>...", cmd_write),
```

Comment:
These subcommands must be added in alphabetical order. Otherwise tab key may not work correctly.

Reply:
Fixed, thank you!